### PR TITLE
Make webflux compatible with latest spring 6

### PR DESCRIPTION
**Pull request by amarziali**
# What Does This Do

Originally created to correctly propagate the span scope to the handlers (since the mono wasn't wrapped hence whatever span created starting from a reactive handler was wrongly attached to the local root).

It also fixes:
* the dispatcher advices that were muzzling around latest spring webflux 6 versions because `Request.getMethodValue` has been deprecated and removed.  Consequences are mostly bad resource name on the handler and missing http route.
* LatestDep webflux spring 6 tests there were no more compatible with that runtime.
* Muzzle was not running on webflux 6. Advices have been migrated from webflux-5 to webflux-6 module since the only linkage issues concerned the resolution of webclient http status code, done with method handles (was already the case for rawStatusCode)

# Motivation

# Additional Notes

This PR enables also the latestDep tests to run against a JDK 17 for obvious reasons. Also, the webflux-6 can be deleted if the test suites are moved on the webflux-5 module. But this can be done in a separate PR since there is no urgency.

Jira ticket: [APMS-10847]

[APMS-10847]: https://datadoghq.atlassian.net/browse/APMS-10847?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/testing/src/main/groovy/datadog/trace/agent/test/asserts/TagsAssert.groovy [hunk 1]
```diff
@@ -107,7 +107,16 @@ class TagsAssert {
...
-    tag("error.type", errorType.name)
+    tag("error.type", {
+      try {
+        if (errorType.isAssignableFrom(Class.forName(it, false, getClass().getClassLoader()))) {
+          return true
+        }
+      } catch (Throwable t) {
+        // can happen
+      }
+      return it == errorType.getName()
```
Comment (mcculls) [inline dd-java-agent/testing/src/main/groovy/datadog/trace/agent/test/asserts/TagsAssert.groovy:118]: Could we check the name first to avoid having to do any extra class-loading when it's identical, for example:

```suggestion
      if (it == errorType.name) {
        return true
      }
      try {
        // also accept type names which are sub-classes of the given error type
        return errorType.isAssignableFrom(
            Class.forName(it as String, false, errorType.class.classLoader))
      } catch (Throwable ignore) {
        return false
      }
```

File: dd-java-agent/instrumentation/spring-webflux-6/src/test/groovy/dd/trace/instrumentation/springwebflux6/client/SpringWebfluxHttpClientBase.groovy [hunk 2]
```diff
@@ -61,17 +67,17 @@ abstract class SpringWebfluxHttpClientBase extends HttpClientTest implements Tes
...
+  Object parentSpan,
+  String method = "GET",
+  boolean renameService = false,
+  boolean tagQueryString = false,
+  URI uri = server.address.resolve("/success"),
+  Integer status = 200,
+  boolean error = false,
+  Throwable exception = null,
+  boolean ignorePeer = false,
+  Map<String, Serializable> extraTags = null) {
```
Comment (mcculls) [inline dd-java-agent/instrumentation/spring-webflux-6/src/test/groovy/dd/trace/instrumentation/springwebflux6/client/SpringWebfluxHttpClientBase.groovy:80]: unrelated formatting change?

File: dd-java-agent/instrumentation/spring-webflux-6/build.gradle [hunk 3]
```diff
@@ -2,15 +2,6 @@ ext {
...
 }
 
-muzzle {
-  pass {
-    group = 'org.springframework'
-    module = 'spring-webflux'
-    versions = "[6,)"
-    javaVersion = "17"
-  }
-}
```
Comment (mcculls) [inline dd-java-agent/instrumentation/spring-webflux-6/build.gradle:12]: Shouldn't we still check muzzle is working as expected?

**Review by mcculls — APPROVED** [review]

File: dd-java-agent/instrumentation/spring-webflux-6/build.gradle (hunk 3 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/spring-webflux-6/build.gradle:12]: I moved that on webflux 5 module. There are no more instrumenter classes here. Have I misunderstood the way it should be?

Comment (mcculls) [reply dd-java-agent/instrumentation/spring-webflux-6/build.gradle:12]: Ah I see, the `spring-webflux-6` module is now for testing purposes only - it has no instrumentations itself

In that case yes, removing this is correct

File: dd-java-agent/instrumentation/spring-webflux-6/src/test/groovy/dd/trace/instrumentation/springwebflux6/client/SpringWebfluxHttpClientBase.groovy (hunk 2 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/spring-webflux-6/src/test/groovy/dd/trace/instrumentation/springwebflux6/client/SpringWebfluxHttpClientBase.groovy]: reverted thanks
